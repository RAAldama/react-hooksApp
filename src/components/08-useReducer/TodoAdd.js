import React from "react";
import { useForm } from "../../hooks/useForm";

const TodoAdd = ({ handleAddTodo }) => {

    const [{ description }, handleInputChange, reset] = useForm({description: ''});

    const handleSubmit = (e) => {
        e.preventDefault();

        if(description.trim().length <= 1){
            return;
        }

        const newTodo = {
            id: new Date().getTime(),
            desc: description,
            done: false
        }

        handleAddTodo(newTodo);
        reset();
    }

  return (
    <>
      <h4>Agregar Todo</h4>
      <hr />

      <form onSubmit={handleSubmit}>
        <input
          type="text"
          name="description"
          placeholder="Tarea a realizar"
          autoComplete="off"
          className="form-control"
          value={description}
          onChange={handleInputChange}
        />

        <div className="d-grid">
          <button type="submit" className="btn btn-outline-info mt-1 btn-block">
            Agreagr
          </button>
        </div>
      </form>
    </>
  );
};

export default TodoAdd;
