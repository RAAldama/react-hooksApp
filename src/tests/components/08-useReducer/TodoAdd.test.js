import { shallow } from "enzyme"
import TodoAdd from "../../../components/08-useReducer/TodoAdd"

describe('tests on <TodoAdd />', () => {
    
    const handleAddTodo = jest.fn();
    const wrapper = shallow(<TodoAdd handleAddTodo={handleAddTodo}/>)

    test('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should not call handleAddTodo', () => {
        const formSubmit = wrapper.find('form').prop('onSubmit');
        formSubmit({ preventDefault(){} });  

        expect(handleAddTodo).toHaveBeenCalledTimes(0);
    })
    
    test('should call handleAddTodo', () => {
        const value = 'OwO UwU';
        const input = wrapper.find('input');
        input.simulate('change', {target: {value, name: 'description'}} );

        const formSubmit = wrapper.find('form').prop('onSubmit');
        formSubmit({ preventDefault(){} });

        expect(handleAddTodo).toHaveBeenCalled();
        expect(handleAddTodo).toHaveBeenCalledWith(expect.any(Object));
        expect(handleAddTodo).toHaveBeenCalledWith({
            id: expect.any(Number),
            desc: value,
            done: false
        });
        expect(wrapper.find('input').prop('value')).toBe('');
        
    })
        
})
