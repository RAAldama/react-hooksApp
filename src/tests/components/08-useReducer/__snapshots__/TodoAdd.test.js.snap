// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <TodoAdd /> should render correctly 1`] = `
<Fragment>
  <h4>
    Agregar Todo
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Tarea a realizar"
      type="text"
      value=""
    />
    <div
      className="d-grid"
    >
      <button
        className="btn btn-outline-info mt-1 btn-block"
        type="submit"
      >
        Agreagr
      </button>
    </div>
  </form>
</Fragment>
`;
